src/main/java/org/craftercms/engine/service/context/SiteContext.java

```diff
@@ -91,6 +91,7 @@
     protected GraphQLFactory graphQLFactory;
     protected SiteCacheWarmer cacheWarmer;
 
+    protected CountDownLatch initializationLatch = new CountDownLatch(1);
```

Comment:
Can you put the `new CountDownLatch(1)` in the constructor please? Normally all classes in Engine instance new objects in their constructors.